### ISSUE TITLE
Rule proposal: return-in-conditionals

### ISSUE THREAD

**Issue by jfmengels** [jfmengels/eslint-plugin-fp#12]
Conditionals (in functions) that don't return a value or end the function are most likely causing a side-effect.
In the case when it is used to assign a value to a variable, in order to not have a complex ternary expression, should probably be moved to a separate function.

Switch cases should also return statement and should therefore not use `break`.

Top-level conditionals should be exempted from this rule, as they can't return values. (bikeshedding here?)

## Invalid

```js
function bar(a) {
  let res;
  if (expression) {
    res = 2;
  } else {
    res = 3;
  }
  return foo(res);
}
```

## Valid

```js
function foo(a) {
  if (!a) {
    return 2;
  }
  return 1;
}

// Top-level conditionals are okay
if (FOO) {
  foo();
} else {
  bar();
}
```

**Comment by jfmengels** [comment]
Maybe worth splitting the rule into two. One for `if` and one for `switch case`

**Comment by justgage** [comment]
+1
